src/pages/test-report/cat-a-mod1/components/speed-check/speed-check.ts

```diff
@@ -149,13 +148,18 @@ export class SpeedCheckComponent {
   }
 
   formatSpeedAttempt = (event: any): number | undefined => {
-    if (!this.speedCheckValidator.pattern.test(event.target.value)) {
-      event.target.value = event.target.value
-        .replace(leadingZero, '')
-        .replace(nonNumericValues, '')
-        .substring(0, Number(this.speedCheckValidator.maxLength));
+    let output: number | undefined;
+    if (event.target.value === '') {
+      output = undefined;
+    } else {
+      if (!this.speedCheckValidator.pattern.test(event.target.value)) {
+        event.target.value = event.target.value
+          .replace(nonNumericValues, '');
+      }
```

Comment:
could condense this by a few lines by removing the else statement, the output variable, returning undefined in the first if & then returning the result of Number(event.target.value) underneath